src/strands/types/_events.py

```diff
@@ -373,4 +374,90 @@ def __init__(self, reason: str | Exception) -> None:
 
 class AgentResultEvent(TypedEvent):
     def __init__(self, result: "AgentResult"):
-        super().__init__({"result": result})
+        super().__init__({"agent_result": True, "result": result})
+
+
+class MultiAgentResultEvent(TypedEvent):
+    """Event emitted when multi-agent execution completes with final result."""
+
+    def __init__(self, result: "MultiAgentResult") -> None:
+        """Initialize with multi-agent result.
+
+        Args:
+            result: The final result from multi-agent execution (SwarmResult, GraphResult, etc.)
+        """
+        super().__init__({"multi_agent_result": True, "result": result})
```

Comment:
Re naming of `result` - [we explicitly document this as being the `AgentResult`](https://strandsagents.com/latest/documentation/docs/user-guide/concepts/streaming/overview/#lifecycle-events) and thus we have people expecting events to have that field of that type - for that reason I don't think we should name this `result`.  

Possible ways forward:

 - name it `multi_agent_result` and get rid of the boolean; con:  inconsistent compared to the other multi-events
 - name it `multi_result`; con: odd nam
 - name it something else